src/v2/Apps/ViewingRoom/Components/ViewingRoomCarousel.tsx

```diff
@@ -1,42 +1,45 @@
 import React, { useState } from "react"
-import { createFragmentContainer, graphql } from "react-relay"
-import { ViewingRoomCarousel_artwork } from "v2/__generated__/ViewingRoomCarousel_artwork.graphql"
 import { Carousel } from "v2/Components/Carousel"
 import { flowRight } from "lodash"
 
 import {
   Box,
   ChevronIcon,
   Flex,
-  Image,
   ProgressBar,
   breakpoints,
 } from "@artsy/palette"
 
 interface ViewingRoomCarouselProps {
-  artwork: ViewingRoomCarousel_artwork
+  data: ReadonlyArray<any>
+  render: (slide: any, slideIndex: number) => JSX.Element
+  height: number[] | number
+  maxWidth?: string | number
+  justifyContent?: string
 }
 
-const ViewingRoomCarousel: React.FC<ViewingRoomCarouselProps> = ({
-  artwork: { title, images },
+export const ViewingRoomCarousel: React.FC<ViewingRoomCarouselProps> = ({
+  data,
+  render,
+  height,
+  maxWidth = breakpoints.lg,
+  justifyContent = "center",
 }) => {
   const computeScrollPercent = selectedIndex =>
-    ((selectedIndex + 1) / images.length) * 100
+    ((selectedIndex + 1) / data.length) * 100
   const [scrollPercent, setScrollPercent] = useState(computeScrollPercent(0))
   const update = flowRight(setScrollPercent, computeScrollPercent)
-  const showProgressBar = images.length > 1
-
-  const CarouselHeight = [350, 550]
+  const showProgressBar = data.length > 1
 
   return (
     <Box width="100%">
       <Flex
-        height={CarouselHeight}
-        maxWidth={breakpoints.lg}
+        height={height}
+        maxWidth={maxWidth}
         m="auto"
         my={2}
         position="relative"
-        justifyContent="center"
+        justifyContent={justifyContent}
```

Comment:
👌

Reply:
Thanks for all your comments! Really helped direct me refine my ts skills. 🙏 🙏

Reply:
np!